Xamarin.Forms.Core/WeakEventManager.cs

```diff
@@ -0,0 +1,159 @@
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+
+namespace Xamarin.Forms
+{
+	internal class WeakEventManager
```

Comment:
It comes from here? There's a WeakEventManager in WindowsBase, but we can't use that because it's not available in PCL. So I wrote a PCL version.